src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -56,43 +65,86 @@ public void setFrameworkId(FrameworkID frameworkId) throws InterruptedException,
     zkState.store(value).get();
   }
 
-  public List<String> getDeadJournalNodes() {
+  private void removeDeadJournalNodes() {
+    deadJournalNodeTimeStamp = null;
     HashMap<String, String> journalNodes = getJournalNodes();
-    Set<String> journalHosts = journalNodes.keySet();
-    List<String> deadJournalHosts = new ArrayList<>();
-
-    for (String journalHost: journalHosts) {
-      if (journalNodes.get(journalHost) == null) {
-        deadJournalHosts.add(journalHost);
-      }
+    List<String> deadJournalHosts = getDeadJournalNodes();
+    for (String deadJournalHost : deadJournalHosts) {
+      journalNodes.remove(deadJournalHost);
     }
-    return deadJournalHosts;
+    setJournalNodes(journalNodes);
   }
 
-  public List<String> getDeadNameNodes() {
+  private void removeDeadNameNodes() {
+    deadNameNodeTimeStamp = null;
     HashMap<String, String> nameNodes = getNameNodes();
-    Set<String> nameHosts = nameNodes.keySet();
-    List<String> deadNameHosts = new ArrayList<>();
-
-    for (String nameHost : nameHosts) {
-      if (nameNodes.get(nameHost) == null) {
-        deadNameHosts.add(nameHost);
-      }
+    List<String> deadNameHosts = getDeadNameNodes();
+    for (String deadNameHost : deadNameHosts) {
+      nameNodes.remove(deadNameHost);
+      log.info("Removing NN Host: " + deadNameHost);
```

Comment:
Nit: Why log removing NN hosts, but not JN/DNs? I would lean towards logging all/none.